### PR TITLE
fix(oas): operation-paramaters outputs better messages

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/spectral/issues/1235

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/rulesets/oas/functions/oasOpParams.ts [hunk 1]
```diff
@@ -1,138 +1,69 @@
...
+    if (param.in in count) {
+      count[param.in]++;
+    }
+  }
 
-        if (Object.keys(inBody).length > 1) {
-          let message = 'Operation has multiple instances of the `in:body` parameter.\n\nParameters found at:\n';
+  if (hasDuplicateParams) {
+    results.push({
+      message: 'Operation must have unique `name` + `in` parameters',
```
Comment (nulltoken) [inline src/rulesets/oas/functions/oasOpParams.ts:53]: @P0lip Couldn't we point to the first duplicated parameter and issue an error message like

```
A parameter in this operation already exposes the same combination of `name` and `in` values.
```

@rossmcdonald  Thoughts?

File: src/rulesets/oas/functions/oasOpParams.ts [hunk 2]
```diff
@@ -1,138 +1,69 @@
...
-          results.push({
-            message,
-            path: ['paths', path, operation],
-          });
-        }
-      }
-    }
+  if (count.body > 1) {
+    results.push({
+      message: 'Operation has multiple instances of the `in:body` parameter',
```
Comment (nulltoken) [inline src/rulesets/oas/functions/oasOpParams.ts:65]: @P0lip And maybe same here, point at the second `in:body` parameter?

File: src/rulesets/oas/functions/oasOpParams.ts (hunk 1 above)
Comment (P0lip) [reply src/rulesets/oas/functions/oasOpParams.ts:53]: We'd need to list all duplicate parameters in this case, wouldn't we?
I thought of it, and both ideas sound good to me.
I'll let @philsturgeon and @rossmcdonald decide.

Comment (nulltoken) [reply src/rulesets/oas/functions/oasOpParams.ts:53]: >We'd need to list all duplicate parameters in this case, wouldn't we?

@P0lip You're right. This is more or less how `oasPathParams` work.  Basically when 3 entries are considered as duplicates, the two last ones are returned as "duplicated" results.

Comment (P0lip) [reply src/rulesets/oas/functions/oasOpParams.ts:53]: Let's do it in the same way then to be consistent.

**Pull request merged**